feat: Add velero release 1.7.1

blackpiglet opened the pull request:
1. port existing 1.6.3 function to 1.7.1
2. update upstream file to velero 1.7.1
3. add restic support
4. update repos

Signed-off-by: Xun Jiang <jxun@vmware.com>

## What this PR does / why we need it
Add Velero release v1.7.1. Add restic supporting function in v1.7.1

## Details for the Release Notes (PLEASE PROVIDE)

```release-note
1. port existing 1.6.3 function to 1.7.1
2. add restic support in 1.7.1
```

## Which issue(s) this PR fixes

Fixes: #2663

## Describe testing done for PR
Create standalone cluster in Docker. 

GINKGO_FOCUS='Basic' GITHUB_TOKEN=$GITHUB_TOKEN REGISTRY_CREDENTIAL_FILE=/Users/jxun/.docker/config.json make e2e-test
GINKGO_FOCUS='Snapshot' GITHUB_TOKEN=$GITHUB_TOKEN REGISTRY_CREDENTIAL_FILE=/Users/jxun/.docker/config.json make e2e-test

## Special notes for your reviewer
None

Commit:
feat: Add velero release 1.7.1

1. port existing 1.6.3 function to 1.7.1
2. update upstream file to velero 1.7.1
3. add restic support
4. update repos

Signed-off-by: Xun Jiang <jxun@vmware.com>

Commit:
Resolve github check error

remove trailing whitespace in MD and yaml file.

Signed-off-by: Xun Jiang <jxun@vmware.com>

seemiller reviewed (changes requested):
Looks good!

I did push this up to the TCE Harbor project and deployed it to my cluster.

Can you update the package yaml to include this image:
```
projects.registry.vmware.com/tce/velero@sha256:fe89fb0897df4f9df2665c305909c6cc5cda42bacf2fbfc2e111764e0431baf8
```

Commit:
Update Velero package image address.

Signed-off-by: Xun Jiang <jxun@vmware.com>

blackpiglet commented:
@seemiller 
Thanks, Nick! 
I updated the package image bundle. Please take another look when you have time.

dvonthenen commented:
going to merge based on this comment (https://github.com/vmware-tanzu/community-edition/pull/2700#pullrequestreview-830760417)... should we need an update, we can do on a follow up PR

seemiller commented on addons/packages/velero/1.7.1/package.yaml:
```diff
@@ -0,0 +1,415 @@
+              default: {}
+            labels:
+              type: object
+              additionalProperties: array
```
```suggestion
              additionalProperties:
                type: array
                items:
                  type: string
```

blackpiglet replied:
Please correct me, if I am wrong. 
In this way, it use map[string]string to represent "labels"
  additionalProperties:
     type: string

By that way, it use map[[]string]string to represent "labels"
              additionalProperties:
                type: array
                items:
                  type: string

I think maybe using first option is more reasonable.

seemiller replied:
@blackpiglet If you feel that's correct, then make your change. Mine was just a suggestion.

seemiller commented on addons/packages/velero/1.7.1/package.yaml:
```diff
@@ -0,0 +1,415 @@
+              default: velero
+            annotations:
+              type: object
+              additionalProperties: string
```
```suggestion
              additionalProperties:
                type: string
```

blackpiglet replied:
Thanks. It's reasonable to use string map to represent the annotations.

davidvonthenen committed:
Update addons/packages/velero/1.7.1/package.yaml

approved from @blackpiglet

Co-authored-by: Nicholas Seemiller <seemiller@gmail.com>

nrb commented on addons/packages/velero/metadata.yaml:
```diff
@@ -9,5 +9,6 @@ spec:
   providerName: VMware
   maintainers:
     - name: Carlisia Thompson
```
We should probably remove Carlisia's name as part of this.

blackpiglet replied:
OK.

Commit:
Modify according to comments

1. modify additionalProperties setting for annotations and labels;
2. modify maintainer list in metadata.

Signed-off-by: Xun Jiang <jxun@vmware.com>

blackpiglet commented:
@nrb @seemiller 
Hi, guys. Could you help to take another look when you have time?

nrb reviewed (approved):
LGTM, but I'm less knowledgeable about ytt than Nick. If all this is doing is bumping the release version and copying it into the new directories, then I'm ok with merging.
